kratos/mpi/sources/mpi_data_communicator.cpp

```diff
@@ -474,6 +474,31 @@ bool MPIDataCommunicator::IsNullOnThisRank() const
     return mComm == MPI_COMM_NULL;
 }
 
+const DataCommunicator& MPIDataCommunicator::GetSubDataCommunicator(
+    const std::vector<int>& rRanks,
```

Comment:
I am thinking that conceptually this should be a `set`. I see that the vector is easier to use, maybe this could be made unique in the DataCommFactory @jcotela ?

Reply:
In the factory a std::vector is used, the same is used here

Reply:
therefore the question :)

Reply:
If I remember correctly, the factory uses a vector just because it passes its data directly as the C array that the corresponding MPI primitive uses. I went with a vector because it seemed straightforward, but I don't have a strong opinion either way. On the other hand, using a set has the nice feature that we are enforcing uniqueness on the interface level, so it is harder to call it with a wrong input.